shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -0,0 +1,425 @@
+   * until {@link #onDestroyView()} is invoked.
+   */
+  @SuppressWarnings("unused")
+  @Nullable
+  public FlutterView getFlutterView() {
+    return flutterView;
+  }
+
+  /**
+   * Starts running Dart within the FlutterView for the first time.
+   *
+   * The difference between the first run and subsequent reloads is that when FlutterView is
+   * run for the first time, it attempts to reuse an existing Dart isolate.
+   */
+  private void doInitialFlutterViewRun() {
+    runFlutterView(getInitialRoute(), getAppBundlePath(), true);
+  }
+
+  /**
+   * Reloads Dart code with that at the given {@code appBundlePath}, sets Flutter's initial route
+   * to {@code initialRoute}, and then reloads the Dart VM.
+   *
+   * TODO(mattcarroll): is the above description accurate?
+   *
+   * @param initialRoute initial Flutter route to display after reloading Dart code
+   * @param appBundlePath Android file path to Flutter's app bundle
+   */
+  public void hotReload(@Nullable String initialRoute, @NonNull String appBundlePath) {
+    runFlutterView(initialRoute, appBundlePath, false);
+  }
+
+  /**
+   * Instructs the FlutterView to start running with the given {@code appBundlePath}, and starting
+   * Flutter at the given {@code initialRoute}.
+   *
+   * @param initialRoute initial Flutter route to display after reloading Dart code
+   * @param appBundlePath Android file path to Flutter's app bundle
+   * @param reuseIsolate whether or not to reuse an existing Dart isolate
+   */
+  private void runFlutterView(@Nullable String initialRoute, @NonNull String appBundlePath, boolean reuseIsolate) {
+    if (initialRoute != null) {
+      flutterView.setInitialRoute(initialRoute);
+    }
+    // TODO: what does it really mean for this condition to be false? is that ok? should we log?
+    if (!flutterView.getFlutterNativeView().isApplicationRunning()) {
+      flutterView.runFromBundle(appBundlePath, null, "main", reuseIsolate);
+    }
+  }
+
+  @Nullable
+  private String getInitialRoute() {
+    return getArguments().getString(ARG_INITIAL_ROUTE);
+  }
+
+  @NonNull
+  private String getAppBundlePath() {
+    // Guaranteed non-null by Fragment factory method which declares the incoming
+    // parameter @NonNull.
+    //noinspection ConstantConditions
+    return getArguments().getString(ARG_APP_BUNDLE_PATH);
+  }
+
+  @Override
+  public final boolean hasPlugin(@NonNull String key) {
+    return flutterView.getPluginRegistry().hasPlugin(key);
+  }
+
+  @Override
+  @Nullable
+  @SuppressWarnings("unchecked")
+  public <T> T valuePublishedByPlugin(@NonNull String pluginKey) {
+    return (T) flutterView.getPluginRegistry().valuePublishedByPlugin(pluginKey);
+  }
+
+  @Override
+  @NonNull
+  public PluginRegistry.Registrar registrarFor(@NonNull String pluginKey) {
+    return flutterView.getPluginRegistry().registrarFor(pluginKey);
+  }
+
+  // TODO: what is the significance of this method? should it be a subclass hook?
```

Comment:
This is for apps who would like to keep the same FlutterIsolate across multiple FlutterViews and yes it would be a subclass hook.

Reply:
Renamed to retainFlutterIsolate() and added docs.